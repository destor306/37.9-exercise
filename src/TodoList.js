import React, {useState} from "react";
import Todo from "./Todo";
import {v4 as uuid} from "uuid"
import NewTodoForm from "./NewTodoForm";


const TodoList = ()=>{

    const [task, setTask] = useState([]);

    const addTask = (newTask) =>{
        setTask(tasks => [...tasks, {...newTask, id: uuid()}])
    }
    const removeTask =(id) =>{
        setTask(tasks => tasks.filter(task=> task.id !== id));
    }
    return(
        <div>
            <h3>TodoList</h3>
            <NewTodoForm addTask={addTask}/>
            {
            task.map(({id, task})=>
            <Todo 
            key={id}
            id={id}
            task={task}
            remove={removeTask}
            />
            )}
        </div>
    )
}

export default TodoList;