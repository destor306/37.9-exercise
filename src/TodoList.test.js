import React from 'react';
import {render, fireEvent} from "@testing-library/react"
import TodoList from './TodoList';


it('renders without crashing', ()=>{
    render(<TodoList />)
})

it('matches snapshot', ()=>{
    const {asFragment} = render(<TodoList />)
    expect(asFragment()).toMatchSnapshot();
})

it('should add new Todo', ()=>{
    const{queryByText, getByLabelText, getByText } = render(<TodoList />)
    const taskInput = getByLabelText('Task')
    const btn = queryByText('Add Task');
    // Fill out form inputs
    fireEvent.change(taskInput, { target: { value: 'Walk Ace' } });

    // Submit form
    fireEvent.click(btn);

    const removeBtn = queryByText("Remove");
    expect(removeBtn).toBeInTheDocument();
    expect(queryByText('Walk Ace')).toBeInTheDocument();
})

it('should delete a Todo', ()=>{
    const{queryByText, getByLabelText } = render(<TodoList />)

    const taskInput = getByLabelText('Task')
    const btn = queryByText('Add Task');
    // Fill out form inputs
    fireEvent.change(taskInput, { target: { value: 'Walk Ace' } });

    // Submit form
    fireEvent.click(btn);

    const removeBtn = queryByText("Remove");
    expect(removeBtn).toBeInTheDocument();
    fireEvent.click(removeBtn);
    expect(removeBtn).not.toBeInTheDocument();
})