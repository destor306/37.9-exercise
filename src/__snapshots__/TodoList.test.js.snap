// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      TodoList
    </h3>
    <form>
      <label
        for="task"
      >
        Task
      </label>
      <input
        id="task"
        name="task"
        type="text"
        value=""
      />
      <button>
        Add Task
      </button>
    </form>
  </div>
</DocumentFragment>
`;
